### PR TITLE
Full text search for resource filtering

### PR DESCRIPTION
**Pull request by harkotha2**
Added new field `searchableExpression` in `searchableData`. searchableExpression contains values of all searchableData fields with colon separated. So now end user can able to search all searchableData fields

```
"searchableData": {
       "kind": "Deployment",
       "name": "clustersubscription",
       "namespace": "razeedeploy",
       "apiVersion": "apps/v1",
       "annotations[\"deployment_kubernetes_io_revision\"]": "1",
       "annotations[\"razee_io_commit_sha\"]": "f83a372fa161b7f0e6c135fee07ce925a6a6d2a7",
       "annotations[\"razee_io_git_repo\"]": "https://github.com/razee-io/ClusterSubscription.git",
       "annotations[\"version\"]": "f83a372fa161b7f0e6c135fee07ce925a6a6d2a7",
       "razeeCommitSha": "f83a372fa161b7f0e6c135fee07ce925a6a6d2a7",
       "searchableExpression": "Deployment:clustersubscription:razeedeploy:apps/v1:1:f83a372fa161b7f0e6c135fee07ce925a6a6d2a7:https://github.com/razee-io/ClusterSubscription.git:f83a372fa161b7f0e6c135fee07ce925a6a6d2a7:f83a372fa161b7f0e6c135fee07ce925a6a6d2a7"
     }

### CODE REVIEW

File: app/apollo/models/resource.schema.js [hunk 1]
```diff
@@ -69,6 +69,13 @@ ResourceSchema.statics.getIds = async(ids)=>{
 ResourceSchema.statics.getByClusterIds = async(clusterIds)=>{
   return await this.find({ cluster_id: { $in: clusterIds } });
 };
+ResourceSchema.index({
+  cluster_id: 'text',
+  selfLink: 'text'
+}, {
+  name: 'cluster_id.selfLink',
+  default_language: 'none'
+});
```
Comment (yingwang-us) [inline app/apollo/models/resource.schema.js:78]: @harkotha2 are we able to do full-text search on searchableData and its fields ?

Comment (harkotha2) [reply app/apollo/models/resource.schema.js:78]: @yingwang-us  - Yes, searchableData fields (`kind , name , namespace`) are will be part `SelfLink`. We should be able to search them

Comment (harkotha2) [reply app/apollo/models/resource.schema.js:78]: As searchableData fields already part of SelfLink, I didn't added searchableData fields to resource index

Screenshot 2020-07-22 at 9 51 07 PM

Comment (yingwang-us) [reply app/apollo/models/resource.schema.js:78]: @harkotha2 there are other fields that are in searchbleData but not in selfLink we would like to search as well if possible ? e.g. ```"searchableData": {
        "kind": "Namespace",
        "name": "xxx-enf",
        "apiVersion": "v1",
        "annotations[\"openshift_io_sa_scc_mcs\"]": "s0:c40,c15",
        "annotations[\"openshift_io_sa_scc_supplemental_groups\"]": "1001590000/10000",
        "annotations[\"openshift_io_sa_scc_uid_range\"]": "1001590000/10000"
      },```

**Comment by mckaymic** [comment]
Can we meet tomorrow morning about this PR?

File: app/apollo/models/resource.schema.js [hunk 2]
```diff
@@ -69,6 +69,7 @@ ResourceSchema.statics.getIds = async(ids)=>{
 ResourceSchema.statics.getByClusterIds = async(clusterIds)=>{
   return await this.find({ cluster_id: { $in: clusterIds } });
 };
+ResourceSchema.index({ cluster_id: 'text', selfLink: 'text', 'searchableData.searchableExpression': 'text' });
```
Comment (yingwang-us) [inline app/apollo/models/resource.schema.js:72]: Hi Harish, could you please move `searchableExpression` as root field of resource, and dump `cluster_id`,the whole json of `searchableData` into searchableExpression, That will allow as search text for Kind, name, api version, subscription_id, and any annotation in searhableData.

Comment (yingwang-us) [reply app/apollo/models/resource.schema.js:72]: nm, might be fine in searchableData.

**Pull request merged**